charts/rancher-thanos/v0.0.1/templates/compact-deployment.yaml

```diff
@@ -0,0 +1,73 @@
+{{- if .Values.thanos.compact.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: thanos-compact-{{ .Release.Name }}
+  labels:
+    app: {{ template "app.name" . }}
+    chart: {{ template "app.version" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: compact
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "app.name" . }}
+      chart: {{ template "app.version" . }}
```

Comment:
Do not use `app.version` as a template metadata label since this is an immutable field and once you bump the chart version it will cause an upgrade error like `selector does not match template labels`. 

notes: global monitoring uses sub-charts design and the templates only contain a CRD YAML file, so bump its parent chart version will not reflect on the sub-charts, also their sub-charts version seems never get updated, do not copy it.